mautic/mautic
# Fixed sendgrid bounce email count for email channel.

When you send an email from an email channel with sendgrid API as service to send mail and if an email bounces from Sendgrid then the email channel details is not able to store on the `lead_donotcontact` table for that email.
Which in result the email channel graph is not able to show the bounce count details in the email channel page.

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging for  3.0 for bug fixes 
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #9045 




#### Description:

Now I am including the channel details to the request data from the mautic side.
So that when Mautic processes the request further, it will be able to store the channel detail in the bounce table. 



#### Steps to test this PR:

1. Use Sendgrid API for "Service to send mail"
2. Create a segment and add some bounced emails
3. Create a email channel with the segment
4. Send a test mail to the contacts
5. After process completed, you are able to see bounce count in the email graph


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9106"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
